sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2ScanRelationPushDown.scala

```diff
@@ -17,61 +17,183 @@
+  }
+
+  def pushdownAggregate(plan: LogicalPlan): LogicalPlan = plan.transform {
+    // update the scan builder with agg pushdown and return a new plan with agg pushed
+    case aggNode @ Aggregate(groupingExpressions, resultExpressions, child) =>
+      child match {
+        case ScanOperation(project, _, sHolder: ScanBuilderHolder) =>
+          sHolder.builder match {
+            case r: SupportsPushDownAggregates =>
+              if (sHolder.builder.asInstanceOf[SupportsPushDownFilters].pushedFilters().length <= 0
+                || r.supportsPushDownAggregateWithFilter()) {
+                if (r.supportsGlobalAggregatePushDownOnly() && groupingExpressions.nonEmpty) {
+                  aggNode // return original plan node
+                } else {
+                  val aggregates = getAggregateExpression(resultExpressions, project, sHolder)
+                  val pushedAggregates = PushDownUtils
+                    .pushAggregates(sHolder.builder, aggregates, groupingExpressions)
+                  if (pushedAggregates.aggregateExpressions.isEmpty) {
+                    aggNode // return original plan node
+                  } else {
+                    // use the aggregate columns as the output columns
+                    // e.g. TABLE t (c1 INT, c2 INT, c3 INT)
+                    // SELECT min(c1), max(c1) FROM t;
+                    // Use min(c1), max(c1) as output for DataSourceV2ScanRelation
+                    // We want to have the following logical plan:
+                    // == Optimized Logical Plan ==
+                    // Aggregate [min(min(c1)#21) AS min(c1)#17, max(max(c1)#22) AS max(c1)#18]
+                    // +- RelationV2[min(c1)#21, max(c1)#22] parquet file ...
+                    val output = aggregates.map {
+                      case agg: AggregateExpression =>
+                        AttributeReference(toPrettySQL(agg), agg.dataType)()
+                    }
+
+                    // No need to do column pruning because only the aggregate columns are used as
+                    // DataSourceV2ScanRelation output columns. All the other columns are not
+                    // included in the output. Since PushDownUtils.pruneColumns is not called,
+                    // ScanBuilder.requiredSchema is not pruned, but ScanBuilder.requiredSchema is
+                    // not used anyways. The schema for aggregate columns will be built in Scan.
+                    val scan = sHolder.builder.build()
+
+                    logInfo(
+                      s"""
+                         |Pushing operators to ${sHolder.relation.name}
+                         |Pushed Aggregate Functions:
+                         | ${pushedAggregates.aggregateExpressions.mkString(", ")}
+                         |Output: ${output.mkString(", ")}
+                      """.stripMargin)
+
+                    val scanRelation = DataSourceV2ScanRelation(sHolder.relation, scan, output)
```

Comment:
shall we check if `output` matches `scan.readSchema`?

Reply:
Actually, I changed the output to be aggregated columns such as `Max(col1), Min(col1)`. I am not sure if I should use `scan.readSchema` for aggregated `output`, because I think schema is for name and type of the table columns. I introduced a new variable `pushedAggSchema`.